### PR TITLE
Track memory usage in Multicast Exchange Node(CTE)

### PR DESCRIPTION
**Pull request by dirtysalt**
ref: #2303  #1675

This PR do following things:
1. rename cpp/header files according to previous reviewer's comment.
2. track memory usage and chunk row size for later spill work.

### CODE REVIEW

File: be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp [hunk 1]
```diff
@@ -150,8 +154,10 @@ void MultiCastLocalExchanger::_update_progress(Cell* fast) {
         Cell* t = _head->next;
         if (t == nullptr) break;
         delete _head;
+        _current_memory_usage -= _head->memory_usage;
```
Comment (trueeyu) [inline be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp:157]: use after free ?

File: be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp [hunk 2]
```diff
@@ -150,8 +154,10 @@ void MultiCastLocalExchanger::_update_progress(Cell* fast) {
...
         _head = t;
     }
+    _current_row_size = _current_accumulated_row_size - _head->accumulated_row_size;
```
Comment (trueeyu) [inline be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp:160]: What is current_row_size used for

File: be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp [hunk 3]
```diff
@@ -28,6 +28,7 @@ MultiCastLocalExchanger::~MultiCastLocalExchanger() {
     while (_head != nullptr) {
         auto* t = _head->next;
         delete _head;
+        _current_memory_usage -= _head->memory_usage;
```
[the remark this answers is not in the document]
Comment (dirtysalt) [reply be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp:31]: Oh, shit. Thanks for pointing out.

File: be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/exec/pipeline/exchange/multi_cast_local_exchange.cpp:160]: How many rows are buffered. I don't know , maybe it's a good indicator for spilling.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 4]
```diff
@@ -167,6 +168,7 @@
     public static final String RUNTIME_JOIN_FILTER_PUSH_DOWN_LIMIT = "runtime_join_filter_push_down_limit";
     public static final String ENABLE_GLOBAL_RUNTIME_FILTER = "enable_global_runtime_filter";
     public static final String ENABLE_COLUMN_EXPR_PREDICATE = "enable_column_expr_predicate";
+    public static final String CTE_MEM_LIMIT = "cte_mem_limit";
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:171]: Do we really need a  session variables?  we should add a new session variables if necessary.

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**